### PR TITLE
FRDQ-105

### PR DESCRIPTION
**Pull request by Toumash**

### CODE REVIEW

**Comment by hayuna** [comment]
<p>Total Coverage: <code>98.33</code></p>
<details><summary>Coverage report</summary>
<p>

File                                      |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
------------------------------------------|----------|----------|----------|----------|-------------------|
All files                                 |    98.36 |    97.67 |    95.24 |    98.33 |                   |
 src                                      |      100 |      100 |      100 |      100 |                   |
  App.tsx                                 |      100 |      100 |      100 |      100 |                   |
  react-app-env.d.ts                      |        0 |        0 |        0 |        0 |                   |
 src/assets                               |        0 |        0 |        0 |        0 |                   |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/Button              |      100 |      100 |      100 |      100 |                   |
  Button.model.ts                         |        0 |        0 |        0 |        0 |                   |
  Button.tsx                              |      100 |      100 |      100 |      100 |                   |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/Header              |      100 |      100 |      100 |      100 |                   |
  Header.tsx                              |      100 |      100 |      100 |      100 |                   |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/Input               |      100 |      100 |      100 |      100 |                   |
  Input.model.ts                          |        0 |        0 |        0 |        0 |                   |
  Input.tsx                               |      100 |      100 |      100 |      100 |                   |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/List                |      100 |      100 |      100 |      100 |                   |
  List.model.ts                           |        0 |        0 |        0 |        0 |                   |
  List.tsx                                |      100 |      100 |      100 |      100 |                   |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/ListItem            |      100 |      100 |      100 |      100 |                   |
  ListItem.model.ts                       |        0 |        0 |        0 |        0 |                   |
  ListItem.tsx                            |      100 |      100 |      100 |      100 |                   |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/Paragraph           |      100 |      100 |      100 |      100 |                   |
  Paragraph.tsx                           |      100 |      100 |      100 |      100 |                   |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/Ranges              |        0 |        0 |        0 |        0 |                   |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/Ranges/DefaultRange |      100 |      100 |      100 |      100 |                   |
  Range.model.ts                          |        0 |        0 |        0 |        0 |                   |
  Range.tsx                               |      100 |      100 |      100 |      100 |                   |
 src/components/atoms/Ranges/LevelRange   |      100 |      100 |      100 |      100 |                   |
  LevelRange.model.ts                     |        0 |        0 |        0 |        0 |                   |
  LevelRange.tsx                          |      100 |      100 |      100 |      100 |                   |
  data.ts                                 |      100 |      100 |      100 |      100 |                   |
 src/components/atoms/SocialButton        |      100 |      100 |      100 |      100 |                   |
  SocialButton.model.ts                   |        0 |        0 |        0 |        0 |                   |
  SocialButton.tsx                        |      100 |      100 |      100 |      100 |                   |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/TextArea            |       75 |      100 |       50 |       75 |                   |
  TextArea.tsx                            |       75 |      100 |       50 |       75 |                19 |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/Toast               |      100 |      100 |      100 |      100 |                   |
  Toast.model.ts                          |        0 |        0 |        0 |        0 |                   |
  Toast.tsx                               |      100 |      100 |      100 |      100 |                   |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/utils               |      100 |        0 |      100 |      100 |                   |
  newid.ts                                |      100 |        0 |      100 |      100 |                 3 |
 src/pages/Home                           |      100 |      100 |      100 |      100 |                   |
  Home.tsx                                |      100 |      100 |      100 |      100 |                   |
  data.ts                                 |      100 |      100 |      100 |      100 |                   |
  index.ts                                |        0 |        0 |        0 |        0 |                   |
 src/styles                               |      100 |      100 |      100 |      100 |                   |
  GlobalStyles.tsx                        |      100 |      100 |      100 |      100 |                   |
  colors.ts                               |      100 |      100 |      100 |      100 |                   |
  devices.ts                              |      100 |      100 |      100 |      100 |                   |
  typography.ts                           |      100 |      100 |      100 |      100 |                   |
</p>
</details>

**Comment by hayuna** [comment]
Is this required to modify 4 files to fix this issue?

**Comment by hayuna** [comment]
<p>Total Coverage: <code>85.50</code></p>
<details><summary>Coverage report</summary>
<p>

File                                         |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
---------------------------------------------|----------|----------|----------|----------|-------------------|
All files                                    |    83.09 |    84.42 |    78.72 |     85.5 |                   |
 src                                         |      100 |      100 |      100 |      100 |                   |
  App.tsx                                    |      100 |      100 |      100 |      100 |                   |
  react-app-env.d.ts                         |        0 |        0 |        0 |        0 |                   |
 src/assets                                  |        0 |        0 |        0 |        0 |                   |
  index.ts                                   |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/Button                 |      100 |      100 |      100 |      100 |                   |
  Button.model.ts                            |        0 |        0 |        0 |        0 |                   |
  Button.tsx                                 |      100 |      100 |      100 |      100 |                   |
  index.ts                                   |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/Header                 |      100 |      100 |      100 |      100 |                   |
  Header.tsx                                 |      100 |      100 |      100 |      100 |                   |
  index.ts                                   |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/Input                  |      100 |      100 |      100 |      100 |                   |
  Input.model.ts                             |        0 |        0 |        0 |        0 |                   |
  Input.tsx                                  |      100 |      100 |      100 |      100 |                   |
  index.ts                                   |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/List                   |      100 |      100 |      100 |      100 |                   |
  List.model.ts                              |        0 |        0 |        0 |        0 |                   |
  List.tsx                                   |      100 |      100 |      100 |      100 |                   |
  index.ts                                   |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/ListItem               |      100 |      100 |      100 |      100 |                   |
  ListItem.model.ts                          |        0 |        0 |        0 |        0 |                   |
  ListItem.tsx                               |      100 |      100 |      100 |      100 |                   |
  index.ts                                   |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/Paragraph              |      100 |      100 |      100 |      100 |                   |
  Paragraph.tsx                              |      100 |      100 |      100 |      100 |                   |
  index.ts                                   |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/Ranges                 |        0 |        0 |        0 |        0 |                   |
  index.ts                                   |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/Ranges/DefaultRange    |      100 |      100 |      100 |      100 |                   |
  Range.model.ts                             |        0 |        0 |        0 |        0 |                   |
  Range.tsx                                  |      100 |      100 |      100 |      100 |                   |
 src/components/atoms/Ranges/LevelRange      |      100 |      100 |      100 |      100 |                   |
  LevelRange.model.ts                        |        0 |        0 |        0 |        0 |                   |
  LevelRange.tsx                             |      100 |      100 |      100 |      100 |                   |
  data.ts                                    |      100 |      100 |      100 |      100 |                   |
 src/components/atoms/SocialButton           |      100 |      100 |      100 |      100 |                   |
  SocialButton.model.ts                      |        0 |        0 |        0 |        0 |                   |
  SocialButton.tsx                           |      100 |      100 |      100 |      100 |                   |
  index.ts                                   |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/TextArea               |       75 |      100 |       50 |       75 |                   |
  TextArea.tsx                               |       75 |      100 |       50 |       75 |                19 |
  index.ts                                   |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/Toast                  |      100 |      100 |      100 |      100 |                   |
  Toast.model.ts                             |        0 |        0 |        0 |        0 |                   |
  Toast.tsx                                  |      100 |      100 |      100 |      100 |                   |
  index.ts                                   |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/utils                  |      100 |        0 |      100 |      100 |                   |
  newid.ts                                   |      100 |        0 |      100 |      100 |                 3 |
 src/components/molecules/SelectInput        |      100 |      100 |      100 |      100 |                   |
  SelectInput.model.ts                       |        0 |        0 |        0 |        0 |                   |
  SelectInput.tsx                            |      100 |      100 |      100 |      100 |                   |
  index.ts                                   |        0 |        0 |        0 |        0 |                   |
 src/components/molecules/SelectInput/Option |    88.24 |       75 |    85.71 |    93.75 |                   |
  Option.model.ts                            |        0 |        0 |        0 |        0 |                   |
  Option.tsx                                 |    88.24 |       75 |    85.71 |    93.75 |                24 |
  index.ts                                   |        0 |        0 |        0 |        0 |                   |
 src/components/molecules/SelectInput/Select |    52.63 |    58.33 |    33.33 |    57.14 |                   |
  Select.model.ts                            |        0 |        0 |        0 |        0 |                   |
  Select.tsx                                 |    52.63 |    58.33 |    33.33 |    57.14 |... 61,65,66,67,68 |
  index.ts                                   |        0 |        0 |        0 |        0 |                   |
 src/hooks                                   |    88.89 |    71.43 |      100 |    88.89 |                   |
  index.ts                                   |        0 |        0 |        0 |        0 |                   |
  useEventListener.ts                        |    92.31 |       70 |      100 |    92.31 |                17 |
  useOutsideClick.ts                         |       80 |       75 |      100 |       80 |                 7 |
 src/pages/Home                              |      100 |      100 |      100 |      100 |                   |
  Home.tsx                                   |      100 |      100 |      100 |      100 |                   |
  data.ts                                    |      100 |      100 |      100 |      100 |                   |
  index.ts                                   |        0 |        0 |        0 |        0 |                   |
 src/styles                                  |      100 |      100 |      100 |      100 |                   |
  GlobalStyles.tsx                           |      100 |      100 |      100 |      100 |                   |
  colors.ts                                  |      100 |      100 |      100 |      100 |                   |
  devices.ts                                 |      100 |      100 |      100 |      100 |                   |
  typography.ts                              |      100 |      100 |      100 |      100 |                   |
</p>
</details>

**Comment by Toumash** [comment]
> Is this required to modify 4 files to fix this issue?

If you only want to have unique ids within an component then a 1 file modification is sufficient, but when you composite multiple components on the page with the same ids thats another error i believe, which my PR fixes

**Comment by hayuna** [comment]
I think if we try to use your `newId` Ids in tests, then if we add another elements in this component/page test will failed, because of different id

**Comment by Toumash** [comment]
> I think if we try to use your `newId` Ids in tests, then if we add another elements in this component/page test will failed, because of different id

Isnt that the purpose of testid in rtl?https://testing-library.com/docs/dom-testing-library/api-queries#bytestid

**Comment by hayuna** [comment]
Right, my bad.

**Comment by hayuna** [comment]
Screenshot 2020-09-26 at 15 25 40
Whole content is generated, so we dont need unique id in page. label->for and input->id, should be the same, but no need to use unique one

File: src/components/atoms/Input/Input.tsx [hunk 1]
```diff
@@ -12,17 +13,20 @@ const Input = ({
   handleOnChange,
   ...props
 }: IProps): JSX.Element => {
-  const renderLabel = () => label && <Label>{label}</Label>;
+  const [inputId] = useState(newId('input-'));
```
Comment (hayuna) [inline src/components/atoms/Input/Input.tsx:16]: We don't need useState here to achieve the same results

```suggestion
  const inputId = newId('input-');
```

File: src/components/atoms/Ranges/DefaultRange/Range.tsx [hunk 2]
```diff
@@ -1,17 +1,19 @@
 import React, { useState } from 'react';
 import { Container, Input } from './Range.styled';
 import { IProps } from './Range.model';
+import newId from '../../utils/newid';
 
 const Range = ({ className }: IProps): JSX.Element => {
   const [rangeValue, setRangeValue] = useState<number>(1);
+  const [id] = useState(newId('range-'));
```
Comment (hayuna) [inline src/components/atoms/Ranges/DefaultRange/Range.tsx:8]: Same here, we can remove this useState support

File: src/components/atoms/Ranges/LevelRange/LevelRange.tsx [hunk 3]
```diff
@@ -10,17 +10,19 @@ import {
 } from './LevelRange.styled';
 import { data } from './data';
 import { IProps } from './LevelRange.model';
+import newId from '../../utils/newid';
```
Comment (hayuna) [inline src/components/atoms/Ranges/LevelRange/LevelRange.tsx:13]: We have support for absolute paths in TS

```suggestion
import newId from 'utils/newid';
```

**Comment by hayuna** [comment]
If we need directory `utils` please create index.ts based on hooks directory, then each exports from utils could be signified like `import { function } from 'utils'`

**Comment by hayuna** [comment]
<p>Total Coverage: <code>85.71</code></p>
<details><summary>Coverage report</summary>
<p>

File                                         |  % Stmts | % Branch |  % Funcs |  % Lines | Uncovered Line #s |
---------------------------------------------|----------|----------|----------|----------|-------------------|
All files                                    |    83.33 |    85.19 |    79.17 |    85.71 |                   |
 src                                         |      100 |      100 |      100 |      100 |                   |
  App.tsx                                    |      100 |      100 |      100 |      100 |                   |
  react-app-env.d.ts                         |        0 |        0 |        0 |        0 |                   |
 src/assets                                  |        0 |        0 |        0 |        0 |                   |
  index.ts                                   |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/Button                 |      100 |      100 |      100 |      100 |                   |
  Button.model.ts                            |        0 |        0 |        0 |        0 |                   |
  Button.tsx                                 |      100 |      100 |      100 |      100 |                   |
  index.ts                                   |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/Header                 |      100 |      100 |      100 |      100 |                   |
  Header.tsx                                 |      100 |      100 |      100 |      100 |                   |
  index.ts                                   |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/Input                  |      100 |      100 |      100 |      100 |                   |
  Input.model.ts                             |        0 |        0 |        0 |        0 |                   |
  Input.tsx                                  |      100 |      100 |      100 |      100 |                   |
  index.ts                                   |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/List                   |      100 |      100 |      100 |      100 |                   |
  List.model.ts                              |        0 |        0 |        0 |        0 |                   |
  List.tsx                                   |      100 |      100 |      100 |      100 |                   |
  index.ts                                   |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/ListItem               |      100 |      100 |      100 |      100 |                   |
  ListItem.model.ts                          |        0 |        0 |        0 |        0 |                   |
  ListItem.tsx                               |      100 |      100 |      100 |      100 |                   |
  index.ts                                   |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/Paragraph              |      100 |      100 |      100 |      100 |                   |
  Paragraph.tsx                              |      100 |      100 |      100 |      100 |                   |
  index.ts                                   |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/Ranges                 |        0 |        0 |        0 |        0 |                   |
  index.ts                                   |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/Ranges/DefaultRange    |      100 |      100 |      100 |      100 |                   |
  Range.model.ts                             |        0 |        0 |        0 |        0 |                   |
  Range.tsx                                  |      100 |      100 |      100 |      100 |                   |
 src/components/atoms/Ranges/LevelRange      |      100 |      100 |      100 |      100 |                   |
  LevelRange.model.ts                        |        0 |        0 |        0 |        0 |                   |
  LevelRange.tsx                             |      100 |      100 |      100 |      100 |                   |
  data.ts                                    |      100 |      100 |      100 |      100 |                   |
 src/components/atoms/SocialButton           |      100 |      100 |      100 |      100 |                   |
  SocialButton.model.ts                      |        0 |        0 |        0 |        0 |                   |
  SocialButton.tsx                           |      100 |      100 |      100 |      100 |                   |
  index.ts                                   |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/Spinner                |      100 |      100 |      100 |      100 |                   |
  Spinner.model.ts                           |        0 |        0 |        0 |        0 |                   |
  Spinner.tsx                                |      100 |      100 |      100 |      100 |                   |
  index.ts                                   |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/TextArea               |       75 |      100 |       50 |       75 |                   |
  TextArea.tsx                               |       75 |      100 |       50 |       75 |                19 |
  index.ts                                   |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/Toast                  |      100 |      100 |      100 |      100 |                   |
  Toast.model.ts                             |        0 |        0 |        0 |        0 |                   |
  Toast.tsx                                  |      100 |      100 |      100 |      100 |                   |
  index.ts                                   |        0 |        0 |        0 |        0 |                   |
 src/components/atoms/utils                  |      100 |        0 |      100 |      100 |                   |
  newid.ts                                   |      100 |        0 |      100 |      100 |                 3 |
 src/components/molecules/SelectInput        |      100 |      100 |      100 |      100 |                   |
  SelectInput.model.ts                       |        0 |        0 |        0 |        0 |                   |
  SelectInput.tsx                            |      100 |      100 |      100 |      100 |                   |
  index.ts                                   |        0 |        0 |        0 |        0 |                   |
 src/components/molecules/SelectInput/Option |    88.24 |       75 |    85.71 |    93.75 |                   |
  Option.model.ts                            |        0 |        0 |        0 |        0 |                   |
  Option.tsx                                 |    88.24 |       75 |    85.71 |    93.75 |                24 |
  index.ts                                   |        0 |        0 |        0 |        0 |                   |
 src/components/molecules/SelectInput/Select |    52.63 |    58.33 |    33.33 |    57.14 |                   |
  Select.model.ts                            |        0 |        0 |        0 |        0 |                   |
  Select.tsx                                 |    52.63 |    58.33 |    33.33 |    57.14 |... 61,65,66,67,68 |
  index.ts                                   |        0 |        0 |        0 |        0 |                   |
 src/hooks                                   |    88.89 |    71.43 |      100 |    88.89 |                   |
  index.ts                                   |        0 |        0 |        0 |        0 |                   |
  useEventListener.ts                        |    92.31 |       70 |      100 |    92.31 |                17 |
  useOutsideClick.ts                         |       80 |       75 |      100 |       80 |                 7 |
 src/pages/Home                              |      100 |      100 |      100 |      100 |                   |
  Home.tsx                                   |      100 |      100 |      100 |      100 |                   |
  data.ts                                    |      100 |      100 |      100 |      100 |                   |
  index.ts                                   |        0 |        0 |        0 |        0 |                   |
 src/styles                                  |      100 |      100 |      100 |      100 |                   |
  GlobalStyles.tsx                           |      100 |      100 |      100 |      100 |                   |
  colors.ts                                  |      100 |      100 |      100 |      100 |                   |
  devices.ts                                 |      100 |      100 |      100 |      100 |                   |
  typography.ts                              |      100 |      100 |      100 |      100 |                   |
</p>
</details>

**Pull request closed without merging**